src/vsr/replica.zig

```diff
@@ -8055,31 +8065,20 @@ pub fn ReplicaType(
             assert(header_head.op <= self.op_prepare_max());
             for (dvcs_all.const_slice()) |dvc| assert(header_head.op >= dvc.header.commit_min);
 
-            // When computing the new commit_max, we cannot simply rely on the fact that our own
-            // commit_min is attached to our own DVC in the quorum.
-            // Consider the case:
-            // 1. Start committing op=N…M.
-            // 2. Send `do_view_change` to self.
-            // 3. Finish committing op=N…M.
-            // 4. Remaining `do_view_change` messages arrive, completing the quorum.
-            // In this scenario, our own DVC's commit is `N-1`, but `commit_min=M`.
-            // Don't let the commit backtrack.
-            const commit_max = @max(
-                self.commit_min,
-                DVCQuorum.commit_max(self.do_view_change_from_all_replicas),
-            );
```

Comment:
Oops, I think you missed my last comment from https://github.com/tigerbeetle/tigerbeetle/pull/2236#discussion_r1731384408! Since `set_op_and_commit_max` takes the @max with self.commit_max, and self.commit_max >= self.commit_min, we should be able to remove this right? Or could self.commit_min be > self.commit_max in some case that I'm missing? 🤯 

https://github.com/tigerbeetle/tigerbeetle/blob/56193862533052b686396f521298920812f75d9d/src/vsr/replica.zig#L8107-L8109

Reply:
Indeed, missed that, thanks! I'd maybe still compute the correct version here, to make it more obvious (or at least add a `maybe(self.commit_max > commit_max)` after we call `set_op_and_commit_max`). 

I am wonder if we should replace that `@max` inside that `set_op_and_commit_max` with an assert, and fix the callsites? But that's for a different PR!

Reply:
Ack, would be good to do that in a separate PR! On that note, I'm wondering if we should come up with a way to keep track of TODOs/ideas from review comments (or if we already have one that I'm not aware of), because it seems like it's easy for them to get lost/forgotten... 

Currently, I'm just keeping track of them in my own Slack chat. For accountability I was wondering if we should keep a master GitHub issue/Slack channel where could dump such TODOs as a team?

Reply:
Yeah, I've been wondering the same re tracking todos...

Reply:
I would love to create a GitHub issue and assign it to myself immediately, but such ideas/TODOs are usually not concrete enough to create a GitHub issue immediately, and this would also lead to too many open issues... 

A single GitHub issue where we go and dump links to conversations from reviews comments doesn't sound too bad to me though!

Reply:
> add a maybe(self.commit_max > commit_max) after we call set_op_and_commit_max

https://github.com/tigerbeetle/tigerbeetle/commit/161ffd0652c247f3d5c4cbe651123a7b03e56251 addresses this.